src/Transport/Utils/AsyncLazy.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.AzureServiceBus.Utils
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+
+    class AsyncLazy<T> : Lazy<Task<T>>
+    {
+        public AsyncLazy(Func<T> valueFactory) :
```

Comment:
This overload is to support value based option. Why for value based code not to use Lazy?